### ISSUE TITLE
It takes too long for Travis to build everything.

### ISSUE THREAD

**Issue by mroberge** [mroberge/hydrofunctions#10]
* HydroFunctions version: 0.1.4
* Python version: 3.4 mostly
* Operating System:

### Description

Travis can take 2-3 minutes to download and build my dependencies, especially Pandas.

### Solution
It might be possible to use conda to build everything. see this:

https://gist.github.com/dan-blanchard/7045057

**Comment by mroberge** [comment]
In addition to building all of the dependencies, I was accidentally calling NWIS in about six! of my tests. I am now skipping these tests and they run much faster. Plus I'm not hammering NWIS every two minutes with the same requests!

Of course, these tests need to be mocked out.

**Comment by mroberge** [comment]
My build went much faster this time! My Python 3.4 builds used to take 9 minutes or so, while Python 3.5 and 3.6 took about 1 minute and 10 seconds.  In commit 07dffff I changed my .travis.yml so that I did a `pip install --upgrade pip setuptools wheel` which updated pip from 9.0.1 to 10.0.1. This helped a little, but probably wasn't necessary. Then, I added `pip install --only-binary=numpy,scipy,pandas,Ipython numpy scipy pandas ipython`, which really sped things up!  With only numpy and scipy installed this way, it took 7 minutes to install, with the whole thing it took about 59 seconds: see [Travis build #167](https://travis-ci.org/mroberge/hydrofunctions/builds/377519489). 

In the future, another possible solution is to use caching to keep binary files that had been successfully built and use them in future builds. More on that here: https://docs.travis-ci.com/user/caching/
